### PR TITLE
developed team app with navbar

### PR DESCRIPTION
**Pull request by upaharika**
Screenshot 2022-12-01 at 12 39 15 AM

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { useState } from "react";
+import { teamDetails } from "../src/assests/teamDetails";
+import CardComponent from "./components/Card";
+import Header from "./components/Header";
+import "./App.css";
+
+const CardContainer = ({ searchedTeamDetails }) => {
```
Comment (nitishnivedan) [inline src/App.js:7]: better to have this CardContainer in a separate file and can import here.

File: src/assests/teamDetails.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
+import male from "../../src/assests/male.jpeg";
+import upaharika from "../../src/assests/upaharika.jpeg";
+import rucha from "../../src/assests/rucha.jpeg";
+import pratik from "../../src/assests/pratik.jpeg";
+import shailendra from "../../src/assests/shailendra.jpeg";
+import shubham from "../../src/assests/shubham.jpeg";
+import abhishek from "../../src/assests/abhishek.png";
+
+export const teamDetails = [
```
Comment (nitishnivedan) [inline src/assests/teamDetails.js:9]: better to have a json file.

File: src/components/Card/index.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import "./style.css";
+
+const CardComponent = ({ member }) => {
```
Comment (nitishnivedan) [inline src/components/Card/index.js:3]: can de structure here further.

const CardComponent = ({ member:{ img, name, designation}})=> (
<div>

...
</div>
)

File: src/components/Card/index.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  return (
+    <div className="card">
+      <img className="member-img" src={member.img}></img>
```
Comment (ap221882) [inline src/components/Card/index.js:6]: img elements must have an alt prop, either with meaningful text, or an empty string for decorative images. 
[jsx-a11y/alt-text](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/tree/HEAD/docs/rules/alt-text.md)

File: src/components/SearchBar/index.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    (member) =>
+      member.name.toLowerCase().includes(searchText) ||
+      member.designation.toLowerCase().includes(searchText)
+  );
+};
+
+const SearchBar = ({ setSearchedTeamDetails }) => {
+  const [searchText, setSearchText] = useState("");
+
+  const onChange = (e) => {
```
Comment (ap221882) [inline src/components/SearchBar/index.js:16]: name your handlers as handleChange, handleSubmit etc. React docs suggest this.

File: src/components/SearchBar/style.css [hunk 6]
```diff
@@ -0,0 +1,13 @@
+.search-bar {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+}
+
+input[type="text"] {
```
Comment (ap221882) [inline src/components/SearchBar/style.css:7]: Can keep your HTML tags/ body, * selectors on top above CSS. Keeping CSS specificity in mind. I know you would have done it here unknowingly, but keep that in mind while writing vanilla CSS.

File: src/components/SearchBar/index.js [hunk 7]
```diff
@@ -0,0 +1,41 @@
+import { useState } from "react";
+import { teamDetails } from "../../assests/teamDetails.js";
+import "./style.css";
+
+const onSearchTeamDetails = (searchText, teamDetails) => {
+  return teamDetails.filter(
+    (member) =>
+      member.name.toLowerCase().includes(searchText) ||
```
Comment (ap221882) [inline src/components/SearchBar/index.js:8]: There is a bug here. You have to make your search text also lowercase before comparing. What if someone searches for "Upaharika"? and "upaharika" doesn't match?

File: public/index.html [hunk 8]
```diff
@@ -0,0 +1,43 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <meta name="theme-color" content="#000000" />
+    <meta
+      name="description"
+      content="Web site created using create-react-app"
```
Comment (ap221882) [inline public/index.html:10]: Change the description of CRA. When you deploy and share, it gives a preview using that.

File: src/assests/teamDetails.js [hunk 9]
```diff
@@ -0,0 +1,90 @@
+import male from "../../src/assests/male.jpeg";
```
Comment (ap221882) [inline src/assests/teamDetails.js:1]: name assests as assets

**Review by ap221882 — COMMENTED** [review]
Nicely composed. Added a few suggestions.

**Review by ap221882 — COMMENTED** [review]
Everything good. Reviewed Assignment 5.

**Review by Pujarini — CHANGES_REQUESTED** [review]
Reviewed. Agree with other reviewers. Great UI.

File: src/App.js [hunk 10]
```diff
@@ -1,20 +1,22 @@
...
+import { teamDetails } from "../src/assests/teamDetails";
 import CardComponent from "./components/Card";
 import Header from "./components/Header";
-import { teamDetails } from "../src/assests/teamDetails";
+import "./App.css";
 
-const CardContainer = () => {
-  return teamDetails.map((member) => {
+const CardContainer = ({ searchedTeamDetails }) => {
+  return searchedTeamDetails.map((member) => {
```
Comment (Pujarini) [inline src/App.js:8]: Too many returns when you can do it in one line.

**Review by Pujarini — CHANGES_REQUESTED** [review]

**Review by nam334 — COMMENTED** [review]
UI looks super cool, kudos!

File: src/components/SearchBar/index.js [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+    const searchedTeamDetails = onSearchTeamDetails(searchText, teamDetails);
+    setSearchedTeamDetails(searchedTeamDetails);
+  };
+  return (
+    <div className="search-bar">
+      <form onSubmit={onInputSubmit}>
+        <input
+          id="restaurant"
+          placeholder="search restaurant"
+          value={searchText}
```
Comment (nam334) [inline src/components/SearchBar/index.js:31]: The placeholder here reads as "search restaurant"...u can change it to something relatable

**Pull request merged**